core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java

```diff
@@ -25,85 +25,164 @@
-        for (IntentObject intentObject : intentObjects) {
-            intent.putExtra(intentObject.reference, intentObject.object);
-        }
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
         context.startActivity(intent);
     }
 
-    public static void jumpToClearingTask(Context context, Class clazz,
-                                          IntentObject... intentObjects) {
+    /**
+     * Sends and intent to start an {@link Activity} for the provided {@link Class} but clearing
+     * the current task and starting a new one.
+     *
+     * @param context      An instance of {@link Context}. Can't be null.
+     * @param clazz        The {@link Class} of the {@link Activity} that will be opened.
+     *                     Can't be null.
+     * @param intentExtras Variable number of instances of {@link IntentExtra} that will be sent
+     *                     as extras to the started {@link Activity}
+     */
+    public static void jumpToClearingTask(@NonNull Context context,
+                                          @NonNull Class clazz,
+                                          @NonNull IntentExtra... intentExtras) {
         Intent intent = new Intent(context, clazz);
-        for (IntentObject intentObject : intentObjects) {
-            intent.putExtra(intentObject.reference, intentObject.object);
+        for (IntentExtra intentExtra : intentExtras) {
+            intent.putExtra(intentExtra.reference, intentExtra.object);
         }
         intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
         context.startActivity(intent);
     }
 
-    public static void jumpToWithAnimation(Activity activity,
-                                           Class clazz,
-                                           ActivityOptionsCompat transitionActivityOptions,
-                                           IntentObject... intentObjects) {
+    /**
+     * Sends and intent to start an {@link Activity} for the provided {@link Class} with an
+     * animation defined by {@link ActivityOptionsCompat}
+     *
+     * @param activity   The {@link Activity} where the intent will be sent from.
+     *                   Can't be null.
+     * @param clazz      The {@link Class} of the {@link Activity} that will
+     *                   be opened. Can't be null.
+     * @param transition An instance of {@link ActivityOptionsCompat} that defines
+     *                   the animation behaviour.
+     */
+    public static void jumpToWithAnimation(@NonNull Activity activity,
+                                           @NonNull Class clazz,
+                                           @NonNull ActivityOptionsCompat transition) {
+        ActivityCompat.startActivity(activity, new Intent(activity, clazz), transition.toBundle());
+    }
+
+    /**
+     * Sends and intent to start an {@link Activity} for the provided {@link Class} with an
```

Comment:
Man, reviewing the Javadocs is next gen reviewing. Good job!

![giphy-downsized](https://cloud.githubusercontent.com/assets/4109119/26694658/904694cc-46de-11e7-8511-2a52138f8f50.gif)